#173 minor updates on doku (#181)
Accenture/reactive-interaction-gateway: docs/tutorial.md

sidebar_label: Tutorial
---

> This tutorial should show a basic usecase for RIG. A frontend (e.g. the mobile app for a chatroom service) will connect to RIG and subscribe to a certain eventtype (e.g. Messages from a chatroom channel). The backend (e.g. chatroom server) will publish the message to RIG and the RIG will do the rest.

Reviewer: ```suggestion
This tutorial shows a basic use case for RIG. A frontend (e.g. the mobile app for a chatroom service) connects to RIG and subscribes to a certain event type (e.g. messages from a chat room). The backend (e.g. chat-room server) publishes the message to RIG, and RIG forwards it to the frontend.
```

> We'll fake frontend/backend using [HTTPie](https://httpie.org/) for HTTP requests, but of course you can also use curl or any other HTTP client. Please note that HTTPie sets the content type to `application/json` automatically, whereas for curl you need to use `-H "Content-Type: application/json"` for all but `GET` requests.

Reviewer: ```suggestion
We simulate frontend and backend HTTP requests using [HTTPie](https://httpie.org/), but of course you can also use curl or any other HTTP client. Please note that HTTPie sets the content type to `application/json` automatically, whereas for curl you need to use `-H "Content-Type: application/json"` for all but `GET` requests.
```

Please take note of the `connection_token` in the CloudEvent's `data` field - you need it in the next step.

## 3. Subscribe to a topic [Frontend]

With the connection established, you can create _subscriptions_ - that is, you can tell RIG which events your app is interested in. RIG needs to know which connection you are referring to, so you need to use the connection token you have noted down in the last step:

```bash
$ CONN_TOKEN="g2dkAA1ub25vZGVAbm9ob3N0AAACrAAAAAAA"

> $ SUBSCRIPTIONS='{"subscriptions":[{"eventType":"chatroom_channel_messages"}]}'

Reviewer: `chatroom_channel_messages` => `chatroom_message`? The event type is often singular and the `channel` in there doesn't seem to add any meaning.

> If there are no subscribers for a received event, the response will still be `202 Accepted` and the event will be silently dropped.

## 5. The event has been delivered to our subscriber [Frontend]

Going back to the first terminal window you should now see your greeting event

# Connect your app to RIG

> In a real-world frontend app the above example to connect your app to RIG would look something like this below.

Reviewer: thought you wanted to get rid of the code below as well.. no longer the case?

Author: I wanted to keep, but for me it was just a bit misleading as it was numbered :)
